runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -303,16 +334,60 @@ public void init(@Nullable final Applet client) throws Exception
 			SwingUtil.setFont(FontManager.getRunescapeFont());
 
 			// Create main window
-			frame = new JFrame();
+			frame = new ClientFrame();
 
 			// Try to enable fullscreen on OSX
 			OSXUtil.tryEnableFullscreen(frame);
 
 			frame.setTitle(properties.getTitle());
 			frame.setIconImage(ICON);
 			frame.getLayeredPane().setCursor(Cursor.getDefaultCursor()); // Prevent substance from using a resize cursor for pointing
-			frame.setLocationRelativeTo(frame.getOwner());
 			frame.setResizable(true);
+			frame.addComponentListener(new ComponentAdapter()
```

Comment:
This should probably be save only when frame is closing, as this is a bit unnecessarily big and we don't really need to save before frame close happens.